controllers/group_user.js

```diff
@@ -0,0 +1,131 @@
+const express = require("express");
+
+const GroupsUsers = require("../models/groups_users");
+const Users = require("../models/users");
+const Groups = require("../models/groups");
+
+const router = express.Router();
+
+const validation = require("../middleware/dataValidation");
+const { groupUsersSchema } = require("../schemas");
+
+router
+	.route("/")
+	.get(async (req, res) => {
+		// check if there are filters present on request body, if so pass filter to find function
+		if (Object.keys(req.body).length > 0) {
+			const groupUsers = await GroupsUsers.find(req.body);
+			res.status(200).json({
+				groupUsers
+			});
+			// if there are no filters being passed on request body, send entire listing of users
+		} else {
+			const groupUsers = await GroupsUsers.find();
+			res.status(200).json({
+				groupUsers
+			});
+		}
+	})
+	.post(validation(groupUsersSchema), async (req, res) => {
+		const { user_id, group_id } = req.body;
+		const user = await Users.find({
+			id: user_id
+		}).first();
+		const group = await Groups.find({
+			id: group_id
+		}).first();
+		if (user && group) {
+			const newGroupUsers = await GroupsUsers.add(req.body);
+			res.status(201).json({
+				newGroupUsers
+			});
+		} else {
+			res.status(404).json({
+				message:
+					"User id provided or Group id provided does not exist, please double check inputs"
+			});
+		}
+	});
+
+// endpoint to retrieve group relationship for logged in user
+router.route("/search").post(async (req, res) => {
+	// check for user_id and group_id in request body
+	const { user_id, group_id } = req.body;
+	// if both user_id and group_id are defined in body move along this branch
+	if (user_id !== undefined || group_id !== undefined) {
+		// find if relation between user and group entered exists, if so return find function from groups_users model
+		const relationExists = await GroupsUsers.find(req.body);
+		console.log("searching for relationship");
+		if (relationExists.length !== 0) {
+			res.status(200).json({
+				relationExists
+			});
+			// if relation does not already exist, check for user and group existence
+		} else {
+			console.log("searching for user and group");
+			const user = await Users.find({
+				id: user_id
+			}).first();
+			const group = await Groups.find({
+				id: group_id
+			}).first();
+			// if user and group exists, send group information
```

Comment:
had this in mind for when a user searches for a group, if he's not a member, then the relation is useless since the group_user table does not have group information other than the id, if there is a relationship we currently have a join on the model to bring in group information - that's my reasoning

Reply:
Aha, so it could just be a group search but it can display different things if the user is a member, admin, etc. and just display the group result if not. That makes a lot of sense, thank you!